src/history.ts

```diff
@@ -0,0 +1,71 @@
+import {WalletBtc} from './service/wallet/wallet.btc';
+import {WalletCreateOptionsInterface} from './service/wallet/common/wallet-create-options.interface';
+import {WalletLtc} from './service/wallet/wallet.ltc';
+import {WalletDash} from './service/wallet/wallet.dash';
+import {WalletZec} from './service/wallet/wallet.zec';
+import {WalletLike} from './service/wallet/common/wallet.interface';
+import {CoinType} from './service/wallet/types/coin.type';
+
+const router = require('express-async-router').AsyncRouter();
+
+router.get('/get_history', async (req: any, res: any) => {
+    const coinType = req.query['coinType'];
+    const page = req.query['page'] || 1;
+    const pageSize = req.query['pageSize'] || 10;
+    const isProd = req.query['isProd'] || false;
+
+    const options: WalletCreateOptionsInterface = {
+        userString: '',
+        isProd: isProd,
+        type: CoinType.BTC,
+        bitcore: null,
+    };
+
+    const wallet = getWallet(coinType, options);
+
+    if (wallet) {
+        const perf = require('execution-time')();
+        perf.start();
+        const result = await wallet.getHistory(page, pageSize, req);
+        const timeResult = perf.stop();
+
+        return res.json({
+            status: 'success',
+            result: result,
+            time: timeResult.time,
+        });
+    } else {
+        return res.json({
+            status: 'error',
+            result: [],
+        });
+    }
+});
+
+function getWallet(coinType: string, options: WalletCreateOptionsInterface) : WalletLike {
+    let wallet: WalletLike =  new WalletBtc(options);
```

Comment:
let wallet: WalletLike =  new WalletBtc(options);
should be 
let wallet: WalletLike;